Library/Homebrew/cleaner.rb

```diff
@@ -153,6 +155,40 @@ def clean_dir(directory)
     end
   end
 
+  sig { void }
+  def rewrite_pkgconfig
+    pc_files = [@formula.lib.realpath, @formula.share.realpath].flat_map do |d|
+      next [] if @formula.skip_clean?(d) || @formula.skip_clean?(d/"pkgconfig")
+
+      d.glob("pkgconfig/*")
+    end
+    return if pc_files.empty?
+
+    keg_only_pc_files = @formula.deps
+                                .reject { |d| d.build? || d.test? }
+                                .map(&:to_formula)
+                                .select(&:keg_only?)
+                                .flat_map { |f| f.prefix.glob("{lib,share}/pkgconfig/*.pc") }
+                                .to_h { |pc_file| [pc_file.basename(".pc").to_s, pc_file.to_s] }
+    keg_only_modules_pattern = keg_only_pc_files.keys.map { |mod| Regexp.escape(mod) }.join("|")
+
+    pc_files.each do |pc_file|
+      any_modification = T.let(false, T::Boolean)
+      lines = pc_file.each_line.map do |line|
+        any_modification ||= line.gsub!(@formula.prefix.realpath.to_s, @formula.opt_prefix.to_s).present?
```

Comment:
One thing is that this doesn't handle multiple copies of formula in Cellar. I guess the more accurate way would be to handle this during link/unlink but that makes logic a bit more complex than always using opt path.

Reply:
I think it's fine to not handle multiple formulae in Cellar.